File: servidor/src/main/java/com/taller/autos/model/SistemaInfo.java
package com.taller.autos.model;

public class SistemaInfo {
    private static SistemaInfo instancia;

    private final String nombreSistema = "Gestión de Vehículos Eléctricos e Híbridos";
    private final String version = "1.0";
    private final String empresa = "Taller 2 - Apps Empresariales";
    private final String[] integrantes = {"Yaser Rondón", "Ismael Cardozo", "Juan Mancipe"};

    private SistemaInfo() {}

    public static SistemaInfo getInstance() {
        if (instancia == null) instancia = new SistemaInfo();
        return instancia;
    }

    public String getNombreSistema() { return nombreSistema; }
    public String getVersion() { return version; }
    public String getEmpresa() { return empresa; }
    public String[] getIntegrantes() { return integrantes; }
}

File: cliente-js/models/SistemaInfo.js
// Patrón Singleton — información global del sistema
export class SistemaInfo {
  static #instancia = null;

  constructor() {
    this.nombreSistema = 'Gestión de Vehículos Eléctricos e Híbridos';
    this.version = '1.0';
    this.empresa = 'Taller 2 - Apps Empresariales';
    this.integrantes = ['Yaser Rondón', 'Ismael Cardozo', 'Juan Mancipe'];
  }

  static getInstance() {
    if (!SistemaInfo.#instancia) {
      SistemaInfo.#instancia = new SistemaInfo();
    }
    return SistemaInfo.#instancia;
  }
}
